dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,176 @@
+to expose the service on ALL nodes in the cluster. This is more wasteful
+of scarce port resources. However, it is a little easier to set up multiple.
+Again, this requires more privilege.
+
+The router is the most common way to access the cluster.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+ExternalIP or NodePort is useful when the HTTP protocol is not being used or
+non-standard ports are in use. There is more manual setup and monitoring
+involved.
+
+The admin needs to set up the external to the cluster networking environment
+so that requests can reach the cluster. For example, names can be configured
+into 
+xref:../install_config/install/prerequisites.html#prereq-dns[DNS]
+to point to specific nodes or other IP addresses in the cluster.
+The DNS wildcard feature can be used to confiugre a subset of names to an
+IP address in the cluster.  This is convenient when using routers because
+it allows the users to set up routes within the cluster without further
+admin attention.
+
+The admin must make sure the local firewall on each node permits the
+request to reach the IP address.
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to make this highly available using replicated services.
+
+endif::[]
+
+
+[[using-a-router]]
+== Using a Router
+
+This is the most common way to access the cluster. A
+xref:../install_config/configuring_routing.adoc#[ router] is configured
+to accept external requests and proxy them based on the configured routes.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+An administrator can create a
+xref:../install_config/install/prerequisites.html#prereq-dns[wild-card DNS]
+entry, and then set up a router, after that the users can self-service
+the ingress without having to bother the admins. The router has controls
+to allow the admin to specify whether the users can self-provision
+hostnames, or if they must fit a pattern the admin defines. The other
+solutions require the admin to do it, or it requires that they delegate
+a lot of privilege.
+
+A set of routes can be created in the various projects. The overall
+set of routes is available to the set of routers. Each router selects
+from the set of routes.  All routers see all routes unless restricted
+by labels on the router, this is called router
+xref:../architecture/core_concepts/routes.adoc:#router-sharding[sharding].
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to configure a router for high availability service using
+multiple replicas.
+
+[[using-the-loadbalancer]]
+== Using a Service of Type LoadBalancer
+
+There are load balancers available on
+xref:../install_config/configuring_aws.adoc#[AWS] and
+xref:../install_config/configuring_gce.adoc#[GCE] clouds as well as
+xref:../admin_guide/tcp_ingress_external_ports.adoc#[non-cloud].
+
```

Comment:
This path does not exist yet, so we'll have to hold off on merging this PR until https://github.com/openshift/openshift-docs/pull/2779 gets in.

However, looks like the path should be:
xref:../admin_guide/tcp_ingress_external_ports.adoc#admin-guide-expose-external-ports[non-cloud].